test/e2e_node/e2e_remote.go

```diff
@@ -0,0 +1,178 @@
+	}
+}
+
+// CreateTestArchive builds the local source and creates a tar archive e2e_node_test.tar.gz containing
+// the binaries k8s required for node e2e tests
+func CreateTestArchive() string {
+	// Build the executables
+	BuildGo()
+
+	// Build the e2e tests into an executable
+	glog.Infof("Building ginkgo k8s test binaries...")
+	testDir, err := getK8sNodeTestDir()
+	if err != nil {
+		glog.Fatalf("Failed to locate test/e2e_node directory %v.", err)
+	}
+	out, err := exec.Command("ginkgo", "build", testDir).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to build e2e tests under %s %v.  Output:\n%s", testDir, err, out)
+	}
+	ginkgoTest := filepath.Join(testDir, "e2e_node.test")
+	if _, err := os.Stat(ginkgoTest); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate test binary %s", ginkgoTest)
+	}
+	defer os.Remove(ginkgoTest)
+
+	// Make sure we can find the newly built binaries
+	buildOutputDir, err := getK8sBuildOutputDir()
+	if err != nil {
+		glog.Fatalf("Failed to locate kubernetes build output directory %v", err)
+	}
+	kubelet := filepath.Join(buildOutputDir, "kubelet")
+	if _, err := os.Stat(kubelet); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate binary %s", kubelet)
+	}
+	apiserver := filepath.Join(buildOutputDir, "kube-apiserver")
+	if _, err := os.Stat(apiserver); os.IsNotExist(err) {
+		glog.Fatalf("Failed to locate binary %s", apiserver)
+	}
+
+	glog.Infof("Building archive...")
+	tardir, err := ioutil.TempDir("", "node-e2e-archive")
+	if err != nil {
+		glog.Fatalf("Failed to create temporary directory %v.", err)
+	}
+	defer os.RemoveAll(tardir)
+
+	// Copy binaries
+	out, err = exec.Command("cp", ginkgoTest, filepath.Join(tardir, "e2e_node.test")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to copy e2e_node.test %v.", err)
+	}
+	out, err = exec.Command("cp", kubelet, filepath.Join(tardir, "kubelet")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to copy kubelet %v.", err)
+	}
+	out, err = exec.Command("cp", apiserver, filepath.Join(tardir, "kube-apiserver")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to copy kube-apiserver %v.", err)
+	}
+
+	// Build the tar
+	out, err = exec.Command("tar", "-zcvf", "e2e_node_test.tar.gz", "-C", tardir, ".").CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to build tar %v.  Output:\n%s", err, out)
+	}
+
+	dir, err := os.Getwd()
+	if err != nil {
+		glog.Fatalf("Failed to get working directory %v.", err)
+	}
+	return filepath.Join(dir, "e2e_node_test.tar.gz")
+}
+
+// RunRemote copies the archive file to a /tmp file on host, unpacks it, and runs the e2e_node.test
+func RunRemote(archive string, host string) (string, error) {
+	// Create the temp staging directory
+	tmp := fmt.Sprintf("/tmp/gcloud-e2e-%d", rand.Int31())
+	_, err := runSshCommand("ssh", host, "--", "mkdir", "-p", tmp)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		output, err := runSshCommand("ssh", host, "--", "rm", "-rf", tmp)
```

Comment:
You are right, the -p option does cause mkdir to return success if the directory already exists.  Since we expect /tmp to already exist, would it be sufficient to remove the "- p" argument?